internal/pkg/runtime/engines/singularity/container.go

```diff
@@ -818,14 +762,14 @@ func (c *container) addOverlayMount(system *mount.System) error {
 				ov.AddLowerDir(filepath.Join(dst, "upper"))
 			}
 
-			err = system.Points.AddImage(mount.PreLayerTag, src, dst, "ext3", flags, imageObject.Offset, imageObject.Size)
+			err = system.Points.AddImage(mount.PreLayerTag, src, dst, "ext3", flags, offset, size)
```

Comment:
```
if err := [...]; err != nil {
    return fmt.Errorf("while [...]: %s", err)
}
```

Also, while we're making changes here we should just apply this to the other clauses in this switch block